File: Source/Rio.Models/DataTransferObjects/Posting/PostingUpsertDto.cs
using System.ComponentModel.DataAnnotations;

namespace Rio.Models.DataTransferObjects.Posting
{
    public class PostingUpsertDto
    {
        [Display(Name = "Type")]
        public int? PostingTypeID { get; set; }
        [Required]
        [Range(1, 100000000)]
        public int? Quantity { get; set; }
        [Required]
        [Range(1, 100000000)]
        public decimal? Price { get; set; }

        public string PostingDescription { get; set; }
        [Display(Name = "Account")]
        [Required]
        public int? CreateAccountID { get; set; }
        [Required]
        public int CreateUserID { get; set; }
    }
}

File: Source/Rio.Web/src/app/shared/models/posting/posting-upsert-dto.ts
export class PostingUpsertDto {
    PostingTypeID: number;
    Price: number;

    Quantity: number;
    PostingDescription: string;
    CreateAccountID: number;
    CreateUserID: number;

    constructor(obj?: any) {
        Object.assign(this, obj);
    }
}